hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_MiscTest.java

```diff
@@ -63,6 +63,25 @@ public void setup() {
         TestProcessors.reset(MEMBER_COUNT * parallelism);
     }
 
+    @Test
+    public void when_jobMetricsDisabled_then_emptyMetrics() throws Throwable {
+        DAG dag = new DAG();
+        dag.newVertex("v1", MockP::new);
+        dag.newVertex("v2", (SupplierEx<Processor>) NoOutputSourceP::new);
+        Job job = jet().newJob(dag);
+
+        //when
```

Comment:
The `// When` comment should be at the place where you do the _when_ action from the method name. Here it's where you disable the job metrics.

Also it's not clear to me whether you disable metrics collection or metrics saving. Looks like collection, but this PR is about metrics saving. Neither is disabled explicitly in this test.